pkg/controller/ps/status.go

```diff
@@ -26,17 +27,33 @@ import (
 	"github.com/percona/percona-server-mysql-operator/pkg/router"
 )
 
-func (r *PerconaServerMySQLReconciler) reconcileCRStatus(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
-	if cr == nil || cr.ObjectMeta.DeletionTimestamp != nil {
-		return nil
-	}
-	if err := cr.CheckNSetDefaults(ctx, r.ServerVersion); err != nil {
-		cr.Status.State = apiv1alpha1.StateError
+func (r *PerconaServerMySQLReconciler) reconcileCRStatus(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL, reconcileErr error) error {
+	if reconcileErr != nil {
+		if cr.Status.State != apiv1alpha1.StateError {
+			meta.SetStatusCondition(&cr.Status.Conditions, metav1.Condition{
+				Status:             metav1.ConditionTrue,
+				Type:               string(apiv1alpha1.StateError),
+				Message:            reconcileErr.Error(),
+				Reason:             "ErrorReconcile",
+				LastTransitionTime: metav1.Now(),
+			})
+
+			cr.Status.Messages = append(cr.Status.Messages, "Error: "+reconcileErr.Error())
+			cr.Status.State = apiv1alpha1.StateError
+		}
+
 		nn := types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}
 		return writeStatus(ctx, r.Client, nn, cr.Status)
 	}
+
 	log := logf.FromContext(ctx).WithName("reconcileCRStatus")
 
+	if cr == nil || cr.ObjectMeta.DeletionTimestamp != nil {
+		return nil
+	}
+
+	cr.Status.Messages = cr.Status.Messages[:0]
```

Comment:
what's the purpose of this?

Reply:
Just reused the logic from PXC, but now that I look at it, I don't see a reason to have it.